### PR TITLE
Replacing Node.rootNode for Node.getRootNode.

### LINKED ISSUE #13699: Implement Node.getRootNode
`Node.rootNode` should be removed as per https://github.com/whatwg/dom/commit/42a45eb0d1af13e510620b6b29696997f0c7aa3d and replaced by [`Node.getRootNode`](https://dom.spec.whatwg.org/#dom-node-getrootnode) as per https://github.com/whatwg/dom/commit/579ef079ded65daa981cac974648621790839072.

Files:
* components/script/dom/node.rs
* components/script/dom/webidls/Node.webidl
* tests/wpt/metadata/dom/historical.html.ini
* tests/wpt/metadata/dom/nodes/rootNode.html.ini
* tests/wpt/metadata/dom/interfaces.ini

### PR DESCRIPTION
**Pull request by Rafagd**
- Changed the specification for the Node element to reflect the new standard.
- Changed the implementation of the NodeMethod.

I'm still waiting for some answers for the questions I've posted on #13699, but I'm already doing the pull request so I can get some feedback as well.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13699 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13703)

### CODE REVIEW

File: components/script/dom/webidls/Node.webidl [hunk 1]
```diff
@@ -30,8 +30,10 @@ interface Node : EventTarget {
 
   [Pure]
   readonly attribute Document? ownerDocument;
+
   [Pure]
-  readonly attribute Node rootNode;
+  Node getRootNode(optional GetRootNodeOptions options);
```
Comment (Ms2ger) [inline components/script/dom/webidls/Node.webidl]: I would prefer if you declared this as `Node getRootNode();`, without the argument. That makes it more obvious that the argument is not implemented yet.

**Review by Ms2ger — CHANGES_REQUESTED** [review]

File: components/script/dom/webidls/Node.webidl (hunk 1 above)
Comment (Rafagd) [inline components/script/dom/webidls/Node.webidl]: ok

**Comment by Ms2ger** [comment]
Code looks good, but I expect we'll need to update some test expectations.

@bors-servo try

**Comment by Rafagd** [comment]
@Ms2ger Is there a way for me to only run the failed tests?

**Comment by KiChjang** [comment]
@Rafagd You need to update the test expectations located in `tests/wpt/metadata/dom/interfaces.html.ini` and `tests/wpt/metadata/dom/nodes/rootNode.html`.

**Comment by Rafagd** [comment]
Oh, now I get it. Working on it.

**Review by KiChjang — CHANGES_REQUESTED** [review]
By default, all tests are expected to PASS. You can simply remove the lines of code where it's expected to FAIL instead of changing them to PASS.

**Comment by Rafagd** [comment]
@KiChjang I'm removing the PASS tests now

**Comment by KiChjang** [comment]
Yes.

**Comment by KiChjang** [comment]
@Rafagd You'll now need to squash all your commits into one, and we're ready to land this!

**Comment by Rafagd** [comment]
Everything is squashed into a single commit now.

**Comment by Ms2ger** [comment]
Thanks!

@bors-servo r+

**Comment by KiChjang** [comment]
@bors-servo retry
* #13360

**Comment by KiChjang** [comment]
@bors-servo retry
* #13775

**Pull request merged**